snapcraft/plugins/gulp.py

```diff
@@ -99,6 +99,13 @@ def build(self):
         env = os.environ.copy()
         env['PATH'] = '{}:{}'.format(
             os.path.join(self._npm_dir, 'bin'), env['PATH'])
+
+        # make npm use http proxy to access registry via https
+        # ref. https://github.com/npm/npm/issues/2050
+        if 'http_proxy' in os.environ:
+            logger.info('Set http_proxy environment variable to "https-proxy"')
```

Comment:
Make this
Setting the "npm_config_https_proxy" environment variable with the value of "https_proxy"